tests/phpt/anon_classes/007_constructor_type_mismatch.php

```diff
@@ -0,0 +1,11 @@
+@kphp_should_fail
+/pass int to argument \$num of anon_class.*::__construct\nbut it's declared as @param string/
+<?php
+
+function test_constructor_type_mismatch() {
+  new class(42) {
+    function __construct(string $num) {}
+  };
+}
+
+test_constructor_type_mismatch();
```

Comment:
Full log of compiler error:
`Compilation error at stage: Calc actual calls, gen by type-inferer.cpp:53
  007_constructor_type_mismatch.php:6  in test_constructor_type_mismatch
    new class(42) {

pass int to argument $num of anon_class\ue076317298712440_0::__construct
but it's declared as @param string

  007_constructor_type_mismatch.php:6  in test_constructor_type_mismatch
    new class(42) {
    42 is int


Compilation terminated due to errors
`